src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/formats/index.js

```diff
@@ -1,8 +1,12 @@
 // @flow
 import {FormatValidator} from 'ajv';
+import bicValidator from './bicValidator';
+import ibanValidator from './ibanValidator';
 import idnEmailValidator from './idnEmailValidator';
 
 const formats: {[string]: FormatValidator} = {
+    'bic': bicValidator,
+    'iban': ibanValidator,
```

Comment:
these are not included in any specification, right? 
if yes, i think we should not add them as format. this will couple the generated json schema to our frontend implementation and make it harder to use in other places (which was the reason for using json schema in the first place) 😕

Reply:
I could just use the `pattern` keyword then for both, would that be better?

Reply:
But a iban can't be fully checked with regex only, because it contains some sort of verification number, so we could only check the iban for proper format, not for validity

Reply:
We could prefix the non-standard formats with e.g. `custom-` or `sulu-`

Reply:
yes i would go with `pattern`. i dont think anybody expects the inputted iban to be checked for validity in depth and it makes the generated schema better while keeping it compatible with other validators beside our frontend 🙂

Reply:
But it would be nice to have instant feedback if you made a typo ... even because it was the same behaviour before

Reply:
Thats true, but we decided to use JSON schema for our validation because it is a standard and we can reuse it in other places. That reason is no longer valid if we start to generate schemas that are not compatible to that standard.

Reply:
I guess most validators could be configured to ignore unknown formats like `iban` or `bic` ... Probably most validators also do not implement `idn-email` (like ajv), therefore we had to implement that by ourselves

Reply:
`idn-email` is part of the specification: https://json-schema.org/understanding-json-schema/reference/string.html#built-in-formats

Reply:
Yes, but still it's not implemented by ajv

Reply:
> I guess most validators could be configured to ignore unknown formats like `iban` or `bic`

Thats possible, but unlike when using `format`, these validators will not validate the data at all then. Also, it does not change the fact that the generated schema does not match the JSON schema specification

Reply:
So if we want to validate data on the server, we need a json schema library which implements `idn-email`, otherwise we would have to implement that by ourselves as well. The exact same applies for `iban`

Reply:
> Yes, but it's still not implemented by ajv

`ajv` is just one implementation that tries to implement the specification. While it is not implemented yet, it probably will be implemented in the future because it is part of the specification.

Reply:
Our json schema isn't valid at all because all object `properties` contain a `name` property ... This is not defined in the spec either

Reply:
If I would not had introduced that definition of the `name` keyword in an earlier pull request, ajv would have thrown a javascript error and would not have validated something at all

Reply:
As already stated above, we decided to generate a JSON schema for validation because it is a standard and we are able to reuse the schema in other places. It took quite an effort to implement things this way compared to just validating things in the frontend. 

Because of this, **I am really against generating schemas that do not match the specification and are coupled to our validator implementation in the frontend.** If we start to do this now for the `iban`, we will do the same thing in other places in the future, If we do that, we just could have implemented validation by ourselves without using a standard. 

> Our json schema isn't valid at all because all object properties contain a name property ... This is not defined in the spec either

> If I would not had introduced that definition of the name keyword in an earlier pull request, ajv would have thrown a javascript error and would not have validated something at all

Then we should improve that and maybe use `title` instead of `name` like in the [specification](https://json-schema.org/draft/2019-09/json-schema-core.html)? That said, this is not a reason for violating the specification in other places.

I think i made my point. I am against violating the specification to slightly improve the validation of an IBAN. If a majority of the @sulu/core-team thinks otherwise, I will obviously accept that 🙂

Reply:
I think we definitely should check the iban for validity, an option would be to use the `pattern` keyword **AND** a custom `iban` format. So probably all json-schema validators will validate the pattern and those we implemented the `iban` format will additionally verify the validity of given iban

Reply:
If we really want to do this, I would be okay with adding an additional `sulu-format` property beside the existing `format` property. But we should prioritize to generate the best schema possible inside of the JSON schema specification before using the `sulu-format` property.

Reply:
I think I would not strictly validate here if there is no clear schema support for iban or bic. Because at the end what is maybe not valid in international use, could be valid in a test system or a simulation software. Like my comment for the url validation, at the end a too strict validation would harm us more then a not so strict validation.
If we really want to add add any patterns we should find a way to make them configurable over the admin configuration.

Reply:
Defining the pattern in the admin configuration is not easily possible, because the json schema for `iban` and `bic` is defined [here](https://github.com/sulu/sulu/blob/2.x/src/Sulu/Bundle/ContactBundle/Resources/js/index.js#L228-L230)

Reply:
fixed